sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+#define OPENTITAN_SW_DEVICE_LIB_CRYPTO_API_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Enum to handle return values of the crypto API.
+ */
+typedef enum crypto_status {
+  // Status is OK; no errors.
+  kCryptoStatusOK = 0x7d3a,
+  // Invalid input arguments; Wrong length or invalid type.
+  kCryptoIncorrectInput = 0x6b7f,
+  // Inconsistencies when cross-checking results, witness, checksums.
+  kCryptoStatusInternalError = 0x4753,
+  // An asynchronous operation is still in progress.
+  kCryptoStatusAsyncIncomplete = 0xe2b7
+} crypto_status_t;
+
+/**
+ * Enum to handle return values of the verification APIs.
+ */
+typedef enum verification_status {
```

Comment:
Should we just use hardened_bool instead?

Reply:
I'm guessing the main reason not to would be that all the types in the API here are self-contained at the moment; using `hardened_bool` would introduce a dependency I'm not sure we want to expose to the OS.